### PR TITLE
Improve java version check

### PR DESCRIPTION
**Pull request by MiniDigger**
misses a proper link to the docs site

tested with these things:
```
D:\IntellijProjects\PaperClean\run>D:\Programms\graalvm-ce-java19-22.3.0\bin\java.exe -jar paper-bundler-1.20.1-R0.1-SNAPSHOT-mojmap.jar      
Starting org.bukkit.craftbukkit.Main
System Info: Java 19 (OpenJDK 64-Bit Server VM 19.0.1+10-jvmci-22.3-b08) Host: Windows 11 10.0 (amd64)

D:\IntellijProjects\PaperClean\run>D:\Programms\graalvm-jdk-20.0.2+9.1\bin\java.exe -jar paper-bundler-1.20.1-R0.1-SNAPSHOT-mojmap.jar
Starting org.bukkit.craftbukkit.Main
System Info: Java 20 (Java HotSpot(TM) 64-Bit Server VM 20.0.2+9-jvmci-23.0-b14) Host: Windows 11 10.0 (amd64)

D:\IntellijProjects\PaperClean\run>C:\Users\Martin\Downloads\openjdk-22-ea+14_windows-x64_bin\jdk-22\bin\java.exe -jar paper-bundler-1.20.1-R0.1-SNAPSHOT-mojmap.jar
Starting org.bukkit.craftbukkit.Main
Unsupported Java detected (22-ea). You are running an unsupported, non official, version. Only general availability versions of Java are supported. Please update your Java version. See https://docs.papermc.io/link-to-faq for more information.

D:\IntellijProjects\PaperClean\run>C:\Users\Martin\Downloads\openjdk-22-ea+14_windows-x64_bin\jdk-22\bin\java.exe -DPaper.IgnoreJavaVersion=true -jar paper-bundler-1.20.1-R0.1-SNAPSHOT-mojmap.jar 
Starting org.bukkit.craftbukkit.Main
Unsupported Java detected (22-ea), but the check was skipped. Proceed with caution! 
System Info: Java 22 (OpenJDK 64-Bit Server VM 22-ea+14-1033) Host: Windows 11 10.0 (amd64)

D:\IntellijProjects\PaperClean\run>C:\Users\Martin\.jdks\corretto-20.0.1\bin\java.exe -jar paper-bundler-1.20.1-R0.1-SNAPSHOT-mojmap.jar
Starting org.bukkit.craftbukkit.Main
System Info: Java 20 (OpenJDK 64-Bit Server VM 20.0.1+9-FR) Host: Windows 11 10.0 (amd64)

D:\IntellijProjects\PaperClean\run>C:\Users\Martin\.gradle\jdks\jdk8u332-b09\bin\java.exe -jar paper-bundler-1.20.1-R0.1-SNAPSHOT-mojmap.jar
Minecraft 1.19 requires running the server with Java 17 or above. Download Java 17 (or above) from https://adoptium.net/

D:\IntellijProjects\PaperClean\run>C:\Users\Martin\.gradle\jdks\eclipse_foundation-16-amd64-windows\jdk-16.0.2+7\bin\java.exe -jar paper-bundler-1.20.1-R0.1-SNAPSHOT-mojmap.jar
Minecraft 1.19 requires running the server with Java 17 or above. Download Java 17 (or above) from https://adoptium.net/

D:\IntellijProjects\PaperClean\run>C:\Users\Martin\Downloads\OpenJDK17U-jre_x64_windows_hotspot_2023-09-08-07-10\jdk-17.0.9+6-jre\bin\java.exe -jar paper-bundler-1.20.1-R0.1-SNAPSHOT-mojmap.jar
Starting org.bukkit.craftbukkit.Main
Unsupported Java detected (17.0.9-beta). You are running an unsupported, non official, version. Only general availability versions of Java are supported. Please update your Java version. See https://docs.papermc.io/link-to-faq for more information.

```

### CODE REVIEW

**Comment by MiniDigger** [comment]
https://github.com/PaperMC/Paperclip/pull/65 is related

**Comment by MiniDigger** [comment]
I cant figure out why the diff is so messed up in the chunk patch, somebody pls handle that on merging, I g2g

**Comment by Machine-Maker** [comment]
> I cant figure out why the diff is so messed up in the chunk patch

It's gonna be something relating to whatever patch/diff tools are being used by paperweight that creates inconsistent (at least with everyone else) diff. A simple apply and rebuild on my machine removed all of them.

**Review by Machine-Maker — APPROVED** [review]

**Review by electronicboy — APPROVED** [review]

File: patches/server/0348-Improve-java-version-check.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
++            boolean skip = Boolean.getBoolean("Paper.IgnoreJavaVersion");
+             float javaVersion = Float.parseFloat(System.getProperty("java.class.version"));
+-            if (javaVersion < 61.0) {
+-                System.err.println("Unsupported Java detected (" + javaVersion + "). This version of Minecraft requires at least Java 17. Check your Java version with the command 'java -version'.");
+-                return;
+-            }
+-            if (javaVersion > 64.0) {
+-                System.err.println("Unsupported Java detected (" + javaVersion + "). Only up to Java 20 is supported.");
++            boolean isOldVersion = javaVersion < 61.0;
++            if (!skip && isOldVersion) {
```
Comment (kennytv) [inline patches/server/0348-Improve-java-version-check.patch:26]: Huh? How would that work?

Comment (lynxplay) [reply patches/server/0348-Improve-java-version-check.patch:26]: > I just added it for code cleaness or whatever, lol  

~ [discord](https://discord.com/channels/289587909051416579/947676006964162611/1149716271408091236)

**Pull request merged**